src/sas/sascalc/file_converter/nxcansas_writer.py

```diff
@@ -359,3 +360,7 @@ def _write_2d_data(self, data, data_entry):
             qy_entry.attrs['resolutions'] = 'dQy'
             dqy_entry = data_entry.create_dataset('dQy', data=data.dqy_data)
             dqy_entry.attrs['units'] = data.Q_unit
+        if data.mask is not None and not all(data.mask == [None]):
```

Comment:
The expression `all(data.mask == [None])` won't work.  

Builtin `all` takes a sequence, such as `all(v is None for v in data.mask)`.  If you give it a numpy array it says
```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

If `data.mask` is a numpy array, it is a pretty funny looking array, containing objects of the form `[None]`.

If `data.mask` is a list, then you don't need `all` since `[None] == [None]` is True and since `all([None] == [None])` produces
```
TypeError: 'bool' object is not iterable
```